scripts/copy_qubit.py

```diff
@@ -72,9 +72,16 @@ def get_data(file_path):
                 read=True
             elif(read and row[sample_index]):
                 #this is every other row
-                data[row[sample_index]]={}
-                data[row[sample_index]]['concentration']=row[conc_index]
-                data[row[sample_index]]['unit']=row[unit_index]
+                if row[sample_index] in data:
+                    #Sample is duplicated, drop the key
+                    log.append("sample {0} has two rows in the Qubit CSV file. Please check the file manually.".format(row[sample_index]))
+                    del data[row[sample_index]]
```

Comment:
The idea is that if there is a double entry, someone messed up with the
qubit experiment, so the log gets updated.
Having a double entry is a human error. Having a triple entry is rather
unlikely, but it couls happen.
Since this is an EPP script, that is only executed with a technician
overlooking the results, they have to rerun the script in case that
happens. I believe handling that case would complicate the conditions a
lot. Do you have a nice way to check that ?

On Fri, Mar 27, 2015 at 3:32 PM, senthil10 <notifications@github.com> wrote:

> In scripts/copy_qubit.py
> <https://github.com/SciLifeLab/genologics/pull/100#discussion_r27298916>:
>
> > @@ -72,9 +72,16 @@ def get_data(file_path):
> >                  read=True
> >              elif(read and row[sample_index]):
> >                  #this is every other row
> > -                data[row[sample_index]]={}
> > -                data[row[sample_index]]['concentration']=row[conc_index]
> > -                data[row[sample_index]]['unit']=row[unit_index]
> > +                if row[sample_index] in data:
> > +                    #Sample is duplicated, drop the key
> > +                    log.append("sample {0} has two rows in the Qubit CSV file. Please check the file manually.".format(row[sample_index]))
> > +                    del data[row[sample_index]]
>
> So you don't want a entry in data at all, for samples with duplicate row
> ? In that case wont a sample appear again in next iteration if it have
> occurred 3 (or odd numbered) times (if that is possible).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/genologics/pull/100/files#r27298916>.
>